packages/wrangler/src/dev/dev.tsx

```diff
@@ -230,6 +230,27 @@ function DevSession(props: DevSessionProps) {
 	useCustomBuild(props.entry, props.build);
 
 	const directory = useTmpDir();
+	const handleError = useErrorHandler();
+
+	// Note: when D1 is out of beta, this (and all instances of `betaD1Shims`) can be removed.
+	const betaD1Shims = useMemo(
```

Comment:
Something later on required referential equality otherwise it spun a render loop. Not sure if that's still the case, but will leave a comment as to its intent.